plugin-default-impl/src/main/java/com/alibaba/nacos/plugin/environment/impl/NacosCustomEnvPluginService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 1999-2021 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.environment.impl;
+
+import com.alibaba.nacos.common.utils.MD5Utils;
+import com.alibaba.nacos.config.server.constant.Constants;
+import com.alibaba.nacos.plugin.environment.spi.CustomEnvironmentPluginService;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Nacos default auth plugin service implementation.
+ *
+ * @author huangtianhui
+ */
+
+public class NacosCustomEnvPluginService implements CustomEnvironmentPluginService {
+
+    @Override
+    public Map<String, Object> customValue(Map<String, Object> property) {
+        String pwd = (String) property.get("db.password.0");
```

Comment:
So only handle `0` db? if I set multiple db like `1`, `2`. Don't solve them?

Reply:
This is just an example, you can select "db.password.0","db.password.1" or any other environment variable to customize

Reply:
Is aimed at environment rather than just the database password